engine/crates/operation-checks/src/aggregate_field_usage.rs

```diff
@@ -0,0 +1,421 @@
+mod async_graphql;
+
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub struct FieldUsage<'a> {
+    pub increment: u64,
+    /// field id -> usage count
+    pub counts: &'a mut HashMap<usize, u64>,
```

Comment:
Might be clearer to name this something like `count_per_field`

Reply:
Good catch, will rename. I am doing a round of documenting and making APIs clearer now (sorry I put this back into draft — feel free to continue to read though, no major change expected).